src/checks/#cc4a#check_in_iteration.clas.abap

```diff
@@ -0,0 +1,347 @@
+      loop at logic_negate-comperasion_operators into data(operator_position).
+        data(token_to_negate) = statement_to_negate-tokens[ operator_position ].
+        if token_to_negate-lexeme = `IS` and statement_to_negate-tokens[ operator_position + 1 ]-lexeme = `NOT`.
+          delete statement_to_negate-tokens index operator_position + 1.
+        elseif token_to_negate-lexeme = `NOT` and statement_to_negate-tokens[ operator_position + 1 ]-lexeme = `IN`.
+          delete statement_to_negate-tokens index operator_position.
+        else.
+          token_to_negate-lexeme = analyzer->negate_comparison_operator( comparison_operator = token_to_negate-lexeme ).
+          statement_to_negate-tokens[ operator_position ] = token_to_negate.
+        endif.
+      endloop.
+    else.
+      insert value #( lexeme = `NOT (` ) into statement_to_negate-tokens index 2.
+      insert value #( lexeme = ` )` ) into statement_to_negate-tokens index lines( statement_to_negate-tokens ) + 1.
+    endif.
+
+    new_statement = statement_to_negate.
+  endmethod.
+
+  method create_quickfix_code_for_if.
+    data(new_statement) = statement.
+    data(analyzer) = /cc4a/abap_analyzer=>create(  ).
+
+    new_statement-keyword = `IF`.
+    new_statement-tokens[ 1 ]-lexeme = `IF`.
+
+    data(flat_new_statement) = analyzer->flatten_tokens( negate_statement( analyzer = analyzer statement = new_statement )-tokens ) && `.`.
+    modified_statement = analyzer->break_into_lines( flat_new_statement ).
+
+    insert `CONTINUE .` into modified_statement index 2.
+    insert `ENDIF .` into modified_statement index 3.
+  endmethod.
+
+  method create_quickfix_code_for_where.
+    data(new_statement) = statement.
+    data(analyzer) = /cc4a/abap_analyzer=>create(  ).
+    data(bool_expression) = ``.
+    data(cut_of_var) = ``.
+    loop at check_statement-tokens assigning field-symbol(<token_information>) where lexeme <> `CHECK`.
+      if contains( val = <token_information>-lexeme sub = variable_name ).
+        cut_of_var = extract_component( token_to_cut_off = <token_information>-lexeme ).
+        bool_expression = bool_expression && ` ` &&  cut_of_var.
+      else.
+        bool_expression = bool_expression && ` ` && <token_information>-lexeme.
+      endif.
+    endloop.
+    data(location_of_var) = find( sub = cut_of_var val = bool_expression ).
+    if location_of_var <> 1.
+      split bool_expression at space into table data(bool_expressions).
+      data(copy_of_expressions) = bool_expressions.
+      data(entries_of_expression) = lines( bool_expressions ).
+      do entries_of_expression times.
+        bool_expressions[ sy-index ] = copy_of_expressions[ entries_of_expression - sy-index + 1 ].
+      enddo.
+      bool_expression = ``.
+      loop at bool_expressions into data(expression).
+        if expression = ` `.
+          continue.
+        endif.
+        bool_expression = bool_expression && ` ` && expression.
+      endloop.
+    endif.
```

Comment:
I think this logic is more complicated than it needs to be: We build `bool_expression` in the first loop but then we search it for the location of `cut_of_var` again and also split it into a table at `space` - couldn't we just have built this table directly in the loop and stored the position, too?